operator/internal/controller/redpanda/redpanda_controller_test.go

```diff
@@ -686,3 +763,59 @@ func TestPostInstallUpgradeJobIndex(t *testing.T) {
 	// `clusterConfigfor` utilizes.
 	require.Equal(t, "bootstrap-yaml-envsubst", job.Spec.Template.Spec.InitContainers[0].Name)
 }
+
+// TestControllerRBAC asserts that the declared Roles and ClusterRoles of the
+// RedpandaReconciler line up with all the resource types it needs to manage.
+func TestControllerRBAC(t *testing.T) {
+	scheme := controller.V2Scheme
+
+	expectedVerbs := []string{"create", "delete", "get", "list", "patch", "update", "watch"}
+
+	roles, err := kube.DecodeYAML(operatorRBAC, scheme)
+	require.NoError(t, err)
+
+	role := roles[1].(*rbacv1.Role)
+	clusterRole := roles[0].(*rbacv1.ClusterRole)
+
+	for _, typ := range redpandachart.Types() {
```

Comment:
Do we have this same test equivalent in the helm chart-defined role? Seems like it'd be nice since we're relying on the generated `role.yaml` in our tests here, but the RBAC rules are actually a part of the operator chart currently.

Reply:
There's a WIP PR from Rafal that tests based off the operator's `role.yaml`: https://github.com/redpanda-data/helm-charts/pull/1593 that I've been patching up to work with these updates. It's a bit less than ideal because the dependency is external to the repo but it'll work for now :/

Reply:
Updated PR here: https://github.com/redpanda-data/helm-charts/pull/1595